gpt_engineer/benchmark/benchmarks/apps/load.py

```diff
@@ -0,0 +1,114 @@
+"""
+Module for loading APPS evaluation tasks.
+
+This module provides functionality to load tasks for evaluating GPT-based models
+on smaller, more focused tasks. It defines a set of tasks with predefined prompts
+and assertions to benchmark the performance of AI models.
+
+Functions
+---------
+load_apps : function
+    Loads the APPS benchmark, which consists of a series coding problems.
+"""
+from collections import OrderedDict
+from pathlib import Path
+from subprocess import TimeoutExpired
+from typing import Union
+
+from datasets import Dataset, DatasetDict, load_dataset, load_from_disk
+
+from gpt_engineer.benchmark.benchmarks.apps.problem import Problem
+from gpt_engineer.benchmark.benchmarks.apps.problems import PROBLEM_IDS
+from gpt_engineer.benchmark.types import Assertable, Benchmark, Task
+from gpt_engineer.core.files_dict import FilesDict
+
+DATASET_PATH = Path("gpt_engineer/benchmark/benchmarks/apps/dataset")
+MAX_N_TEST_EXAMPLES = 10
+
+
+class AppsAssertion:
+    def __init__(self, expected: str, command: str):
+        self.expected_output = self._format(expected)
+        self.command = command
+
+    def evaluate(self, assertable: Assertable) -> bool:
+        pro = assertable.env.popen(self.command)
```

Comment:
This may sound a little counterintuitive to the architecture, but I think it may be preferable to create a fresh DiskExecutionEnvironment, upload the code and run from there, instead of using the global one. The reason is that, executing the code may have side-effects like creating and removing files and folders. If this is the case, multiple executions of the same program with different inputs will not be independent.

Reply:
I see, will do